pkg/webhook/statefulset/statefulset.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statefulset
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/golang/glog"
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap.com/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/client/clientset/versioned"
+	"github.com/pingcap/tidb-operator/pkg/label"
+	"github.com/pingcap/tidb-operator/pkg/webhook/util"
+	"k8s.io/api/admission/v1beta1"
+	apps "k8s.io/api/apps/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+)
+
+var (
+	versionCli   versioned.Interface
+	deserializer runtime.Decoder
+)
+
+func init() {
+	deserializer = util.GetCodec()
+}
+
+func AdmitStatefulSets(ar v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
+
+	name := ar.Request.Name
+	namespace := ar.Request.Namespace
+	glog.Infof("admit statefulsets [%s/%s]", name, namespace)
+
+	setResource := metav1.GroupVersionResource{Group: "apps", Version: "v1beta1", Resource: "statefulsets"}
+	if ar.Request.Resource != setResource {
+		err := fmt.Errorf("expect resource to be %s", setResource)
+		glog.Errorf("%v", err)
+		return util.ARFail(err)
+	}
+
+	if versionCli == nil {
```

Comment:
Could we put the initialization of `versionCli` to `init()` in favor of fail-fast

Reply:
I think it is not neccessry